### PR TITLE
Add support for Kerberos security & NameNode status

### PR DESCRIPTION
**Pull request by natbraun**
This change is [Reviewable](https://reviewable.io/reviews/criteo-forks/tcollector/2)

### CODE REVIEW

**Comment by jbnote** [comment]
+1 but this is wishful thinking :)

File: collectors/0/hadoop_datanode.py [hunk 1]
```diff
@@ -41,7 +42,7 @@ class HadoopDataNode(HadoopHttp):
     """
 
     def __init__(self):
-        super(HadoopDataNode, self).__init__('hadoop', 'datanode', 'localhost', 50075)
+        super(HadoopDataNode, self).__init__('hadoop', 'datanode', socket.gethostname(), 1006)
```
Comment (dopuskh3) [inline collectors/0/hadoop_datanode.py]: Would be nicer to store string and int constants in class variables

File: collectors/0/hadoop_namenode.py [hunk 2]
```diff
@@ -39,15 +33,28 @@ class HadoopNameNode(HadoopHttp):
...
+    def to_numeric(self, context, key, value):
+        if (context, key) == (['fsnamesystem'], 'tag.HAState'):
+            if value == 'standby':
+                return 0
+            elif value == 'active':
+                return 1
+            else:
+                return -1
+        elif (context, key) == (['namenodeinfo'], 'Safemode'):
+            if value == '':
```
Comment (dopuskh3) [inline collectors/0/hadoop_namenode.py:53]: same here.

File: collectors/lib/hadoop_http.py [hunk 3]
```diff
@@ -32,20 +35,18 @@ class HadoopHttp(object):
...
+        self.session = requests.Session()
+        self.auth = HTTPKerberosAuth(mutual_authentication=OPTIONAL)
 
     def request(self):
         try:
-            self.server.request('GET', self.uri)
-            resp = self.server.getresponse().read()
+            r = self.session.get(self.url, auth=self.auth)
+            if r.status_code != 200:
+                raise Exception("Could not get jmx data through HTTP transport: error " + str(r.status_code))
```
Comment (dopuskh3) [inline collectors/lib/hadoop_http.py:46]: Could you use a more specific exception (create a JMXException?)

File: collectors/lib/hadoop_http.py [hunk 4]
```diff
@@ -32,20 +35,18 @@ class HadoopHttp(object):
     def __init__(self, service, daemon, host, port, uri="/jmx"):
         self.service = service
         self.daemon = daemon
-        self.port = port
-        self.host = host
-        self.uri = uri
-        self.server = httplib.HTTPConnection(self.host, self.port)
-        self.server.auto_open = True
+        self.url = "http://" + host + ":" + str(port) + uri
```
Comment (dopuskh3) [inline collectors/lib/hadoop_http.py:38]: NIT: I usually prefer join to + like ''.join([ 'http://' , host, ':', port, uroi])

File: collectors/0/hadoop_nodemanager.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+# will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty
+# of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser
+# General Public License for more details.  You should have received a copy
+# of the GNU Lesser General Public License along with this program.  If not,
+# see <http://www.gnu.org/licenses/>.
+
+import sys
+from collectors.lib.hadoop import Hadoop
+
+REPLACEMENTS = {
```
Comment (dopuskh3) [inline collectors/0/hadoop_nodemanager.py:18]: Unused

File: collectors/0/hadoop_resourcemanager.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+# will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty
+# of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser
+# General Public License for more details.  You should have received a copy
+# of the GNU Lesser General Public License along with this program.  If not,
+# see <http://www.gnu.org/licenses/>.
+
+import sys
+from collectors.lib.hadoop import Hadoop
+
+REPLACEMENTS = {
```
Comment (dopuskh3) [inline collectors/0/hadoop_resourcemanager.py:18]: Unused

File: collectors/0/hadoop_resourcemanager.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+def main(args):
+    rm = Hadoop(replacements=REPLACEMENTS, name='resourcemanager', delay=90, port=8088)
+    return rm.loop()
+
```
Comment (dopuskh3) [inline collectors/0/hadoop_resourcemanager.py:24]: looks like a repeated pattern:

REPLACEMENTS = {}
def main(args):
  rm = Hadoop(...)
  return rm.loop()

I think there is an opportunity to factorize

File: collectors/0/hbase_regionserver.py [hunk 8]
```diff
@@ -30,7 +31,7 @@
 
 class HBaseRegionserver(HadoopHttp):
     def __init__(self):
-        super(HBaseRegionserver, self).__init__("hbase", "regionserver", "localhost", 60030)
+        super(HBaseRegionserver, self).__init__("hbase", "regionserver", socket.gethostname(), 60030)
```
Comment (dopuskh3) [inline collectors/0/hbase_regionserver.py]: String constants

**Pull request closed without merging**